File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/osm/MemberParser.java
package ch.bailu.aat_lib.xml.parser.osm;

import org.xmlpull.v1.XmlPullParser;

import java.io.IOException;

import ch.bailu.aat_lib.coordinates.LatLongE6;
import ch.bailu.aat_lib.xml.parser.scanner.Scanner;
import ch.bailu.util.Objects;

public class MemberParser extends TagParser {

    public MemberParser() {
        this("member");
    }
    public MemberParser(String tag) {
        super(tag);
    }


    @Override
    protected void parseText(XmlPullParser parser, Scanner scanner) throws IOException {

    }

    @Override
    protected void parseAttributes(XmlPullParser parser, Scanner scanner) throws IOException {
        new Attr(parser) {
            @Override
            public void attribute(String name, String value) throws IOException {
                if (Objects.equals(name, "ref")) {
                    scanner.id.scan(value);

                    LatLongE6 point = scanner.referencer.get(scanner.id.getInt());

                    if (point != null) {
                        scanner.referencer.bounding.add(point);
                        scanner.referencer.resolved++;
                    }
                }
            }
        };
    }


    @Override
    protected boolean parseTags(XmlPullParser parser, Scanner scanner) {
        return false;
    }

    @Override
    protected void parsed(XmlPullParser parser, Scanner scanner) {
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/xml/parser/osm/MemberParser.kt
package ch.bailu.aat_lib.xml.parser.osm

import ch.bailu.aat_lib.util.Objects.equals
import ch.bailu.aat_lib.xml.parser.parseAttributes
import ch.bailu.aat_lib.xml.parser.scanner.Scanner
import org.xmlpull.v1.XmlPullParser
import java.io.IOException

open class MemberParser @JvmOverloads constructor(tag: String = "member") : TagParser(tag) {
    @Throws(IOException::class)
    override fun parseText(parser: XmlPullParser, scanner: Scanner) {
    }

    @Throws(IOException::class)
    override fun parseAttributes(parser: XmlPullParser, scanner: Scanner) {
        parser.parseAttributes { name, value ->
            if (equals(name, "ref")) {
                scanner.id.scan(value)
                val point = scanner.referencer[scanner.id.int]
                if (point != null) {
                    scanner.referencer.bounding.add(point)
                    scanner.referencer.resolved++
                }
            }
        }
    }

    override fun parseTags(parser: XmlPullParser, scanner: Scanner): Boolean {
        return false
    }

    override fun parsed(parser: XmlPullParser, scanner: Scanner) {}
}
